sound/soc/sof/trace.c

```diff
@@ -13,6 +13,226 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+#define TRACE_FILTER_ELEMENTS_PER_ENTRY 4
+#define TRACE_FILTER_MAX_CONFIG_STRING_LENGTH 1024
+
+static int trace_filter_append_elem(int32_t key, int32_t value,
+				    struct sof_ipc_trace_filter_elem *elem_list,
+				    int capacity, int *counter)
+{
+	if (*counter >= capacity)
+		return -ENOMEM;
+
+	elem_list[*counter].key = key;
+	elem_list[*counter].value = value;
+	++*counter;
+
+	return 0;
+}
+
+static int trace_filter_parse_entry(struct snd_sof_dev *sdev, const char *line,
+				    struct sof_ipc_trace_filter_elem *elem,
+				    int capacity, int *counter)
+{
+	int len = strlen(line);
+	int cnt = *counter;
+	int log_level;
+	uint uuid_id;
```

Comment:
oh, this is the first time I see this in the kernel... It's defined in types.h along with `uchar`, `ushort`, and `ulong` with a comment "/* sysv */". I see it used a few times under `sound/` and even under `sound/soc/` so, I guess it's ok, but it's kind of an odd ball...

Reply:
changed to uint32_t